discussment-examples/forum/src/main/java/org/danekja/discussment/forum/core/service/imp/DefaultDashboardService.java

```diff
@@ -0,0 +1,152 @@
+package org.danekja.discussment.forum.core.service.imp;
+
+import org.danekja.discussment.core.accesscontrol.dao.NewPermissionDao;
+import org.danekja.discussment.core.accesscontrol.domain.*;
+import org.danekja.discussment.core.domain.Category;
+import org.danekja.discussment.core.domain.Discussion;
+import org.danekja.discussment.core.domain.Topic;
+import org.danekja.discussment.forum.core.domain.User;
+import org.danekja.discussment.forum.core.service.DashboardService;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+public class DefaultDashboardService implements DashboardService {
```

Comment:
See comment for DashboardService in article example. Remove this.

Note: if you need to copy and paste this much code, it usually means something is not done correctly.